### PR TITLE
Remove version from addons

### PR DESCRIPTION
**Pull request by hakman**
... as discussed and agreed during 2021.06.24 Office Hours.

### CODE REVIEW

File: channels/pkg/api/channel.go [hunk 1]
```diff
@@ -72,27 +65,3 @@ type AddonSpec struct {
...
-			if addon.Name != nil {
-				name = *addon.Name
-			}
-
-			_, err := semver.ParseTolerant(*addon.Version)
-			if err != nil {
-				return fmt.Errorf("addon %q has unparseable version %q: %v", name, *addon.Version, err)
-			}
-		}
-		if addon.KubernetesVersion != "" {
```
Comment (johngmyers) [inline channels/pkg/api/channel.go:92]: Please don't remove this check.

**Review by johngmyers — CHANGES_REQUESTED** [review]

File: channels/pkg/api/channel.go (hunk 1 above)
Comment (hakman) [reply channels/pkg/api/channel.go:92]: Done.

**Comment by johngmyers** [comment]
Oh, I have a new issue: how are we going to deal with #11776 and the need to replace instead of update for some transitions, such as changing the ports for coredns?

**Comment by hakman** [comment]
> Oh, I have a new issue: how are we going to deal with #11776 and the need to replace instead of update for some transitions, such as changing the ports for coredns?

Maybe based on kOps version?

**Review by johngmyers — APPROVED** [review]
Actually, we could do it off of the Id.

**Pull request merged**